Create and interpret a confidence interval for the proportion of a chosen categorical
variable. How reliable is this interval for decision-making?

In [2]:
import numpy as np
import pandas as pd

In [3]:
df = pd.read_csv(r'E:\Code\Maths assignment\tested.csv')

In [4]:
# taking out sample
sample = df['Survived'].sample(n=100, random_state=42)

In [5]:
SurvivedCnt = sample[sample == 1].sum()
NotSurvivedCnt = sample[sample == 0].sum()
total=sample.shape[0]

In [6]:
n=total
# survived proportion
p = SurvivedCnt/total
# for 95% CI
z = 1.96

In [7]:
SE = np.sqrt((p * (1 - p)) / n)

In [8]:
CIlower = p - z * SE
CIupper = p + z * SE
CIlower

np.float64(0.2565139582611393)

In [9]:
print(f"({CIlower},{CIupper})")

(0.2565139582611393,0.44348604173886064)


 we estimate with 95% confidence that the true proportion of survivors on the Titanic falls between 25.66% and 44.35%